ide/editor.autosave/src/org/netbeans/modules/editor/autosave/AutoSavePanel.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.editor.autosave;
+
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.lang.ref.WeakReference;
+import java.util.prefs.PreferenceChangeEvent;
+import java.util.prefs.PreferenceChangeListener;
+import javax.swing.BorderFactory;
+import javax.swing.GroupLayout;
+import javax.swing.GroupLayout.Alignment;
+import javax.swing.JCheckBox;
+import javax.swing.JLabel;
+import javax.swing.JSpinner;
+import javax.swing.LayoutStyle.ComponentPlacement;
+import javax.swing.SpinnerNumberModel;
+import org.netbeans.modules.editor.autosave.command.AutoSaveController;
+import org.openide.awt.Mnemonics;
+import org.openide.util.NbBundle;
+
+final class AutoSavePanel extends javax.swing.JPanel {
+
+    AutoSavePanel(final AutoSaveOptionsPanelController controller) {
+        spnModel = new SpinnerNumberModel(10, 0, 999, 1);
+
+        initComponents();
+
+        AutoSaveController.prefs().
+                addPreferenceChangeListener(new WeakReference<PreferenceChangeListener>(evt -> {
```

Comment:
Wait... this just creates a WeakReference instance wrapping the listener, gets the listener, and then immediately throws away the WeakReference?